BrowserKit/Sources/SiteImageView/ImageProcessing/SiteImageFetcher/SiteImageFetcher.swift

```
@@ -7,11 +7,9 @@ import Kingfisher
 
 protocol SiteImageFetcher {
     /// Fetches an image from a specific URL
-    /// - Parameters:
-    ///   - imageURL: Given a certain image URL
-    ///   - completion: The return will be an image or an image error following Result type
-    func fetchImage(imageURL: URL,
-                    completion: @escaping ((Result<UIImage, ImageError>) -> Void))
+    /// - Parameter imageURL: Given a certain image URL
+    /// - Returns: An image or an image error following Result type
+    func fetchImage(from imageURL: URL) async -> Result<UIImage, ImageError>
```

Same comment as above. Things should throw, rather than return a tuple.